### PR TITLE
UTF-8 All The Things

### LINKED ISSUE #1039: Require UTF-8
See https://github.com/whatwg/html/pull/3091

### PR DESCRIPTION
**Pull request by edent**
* Update spec to insist on UTF-8
* Fixes #1039

### CODE REVIEW

File: sections/semantics-document-metadata.include [hunk 1]
```diff
@@ -1417,24 +1414,31 @@
...
   * The character encoding declaration must be serialized without the use of
       <a>character references</a> or character escapes of any kind.
   * The element containing the character encoding declaration must be serialized completely
       within <dfn>the first 1024 bytes</dfn> of the document.
+  * Due to a number of restrictions on <{meta}> elements, there can only be one
+      <code>meta</code>-based character encoding declaration per document.
 
-  In addition, due to a number of restrictions on <{meta}> elements, there can only be one
-  <code>meta</code>-based character encoding declaration per document.
+  Authoring tools should default to using <a>UTF-8</a> for newly-created documents. [[!ENCODING]]
```
Comment (chaals) [inline sections/semantics-document-metadata.include]: If we require utf-8 then this (and similar requirements) have to be a must.

File: sections/semantics-document-metadata.include [hunk 2]
```diff
@@ -1417,24 +1414,31 @@
 
   A <dfn>character encoding declaration</dfn> is a mechanism by which the <a>character encoding</a>
   used to store or transmit a document is specified.
+  
+  The only acceptable character encoding declaration for the modern web is <a>UTF-8</a>.
```
Comment (chaals) [inline sections/semantics-document-metadata.include:1418]: Is it possible to rephrase this less didactically?

File: sections/semantics-document-metadata.include [hunk 3]
```diff
@@ -1449,23 +1453,8 @@
...
-  authors should not use <a>ISO-2022-JP</a>.  [[!ENCODING]]
-
-  <p class="note">
-    Some encodings that are not defined in the WHATWG Encoding specification use bytes in the range 0x20
-    to 0x7E, inclusive, to encode characters other than the corresponding characters in the range
-    U+0020 to U+007E, inclusive, and represent a potential security vulnerability: A user agent
-    might end up interpreting supposedly benign plain text content as HTML tags and JavaScript.
-  </p>
+  <a state for="http-equiv" lt="content-type">encoding declaration state</a>, then the character
+  encoding used must be an <a>ASCII-compatible encoding</a>.
```
Comment (chaals) [inline sections/semantics-document-metadata.include]: why doesn't this just require utf-8?

File: sections/iana.include [hunk 4]
```diff
@@ -37,8 +37,8 @@
       :: The <code>charset</code> parameter may be provided to specify the
           <a>document's character encoding</a>, overriding any
           [=character encoding declarations=] in the document other than a Byte Order Mark (BOM).
-          The parameter's value must be one of the <a lt="character encoding">labels</a> of the <a>character encoding</a>
-          used to serialize the file. [[!ENCODING]]
+          The parameter's value must be an <a>ASCII case-insensitive</a> match for the string
+          "<code>utf-8</code>". [[!ENCODING]]
```
Comment (chaals) [inline sections/iana.include]: If we just enforce a string, there is no normative dependency on the encoding spec here.

**Review by chaals — CHANGES_REQUESTED** [review]
If we are going to make a change requiring UTF-8 from authors, we should ask for very wide review.

Meanwhile, requested a couple of small changes at least to be considered...

File: sections/semantics-document-metadata.include (hunk 2 above)
Comment (edent) [reply sections/semantics-document-metadata.include:1418]: No :-)

That is, I'm not sure how else to phrase it in order to get the point across. Suggestions welcome.

**Comment by edent** [comment]
Regarding wide review.  UTF-8 is now 91% of the web https://w3techs.com/technologies/overview/character_encoding/all

Happy to hear arguments why it shouldn't be mandated.

File: sections/semantics-document-metadata.include (hunk 2 above)
Comment (edent) [reply sections/semantics-document-metadata.include:1418]: > The problems outlined here go away when exclusively using UTF-8, which is one of the many reasons that is now the mandatory encoding for all things.
> https://www.w3.org/TR/2018/CR-encoding-20180327/

File: sections/iana.include [hunk 5]
```diff
@@ -264,8 +264,10 @@
   <dt>Optional parameters:</dt>
   <dd>
     <dl>
-    <dt><code data-x="">charset</code></dt>
-    <dd>The charset parameter may be provided. The parameter's value must be "<code>utf-8</code>". This parameter serves no purpose; it is only allowed for compatibility with legacy servers.</dd>
+      <dt><code data-x="">charset</code></dt>
+      <dd>The charset parameter may be provided. The parameter's value must be "<code>utf-8</code>".
+          This parameter serves no purpose; it is only allowed for compatibility with legacy servers.
```
Comment (chaals) [inline sections/iana.include]: "This parameter is for compatibility with legacy servers", no?

File: sections/semantics-document-metadata.include (hunk 2 above)
Comment (chaals) [reply sections/semantics-document-metadata.include:1418]: yeah, this is hardly a high-order problem.

**Review by chaals — CHANGES_REQUESTED** [review]
This needs to be noted in the changes section.

**Comment by edent** [comment]
@chaals I have updated the Changes file.

**Review by chaals — APPROVED** [review]

**Comment by r12a** [comment]
See comments from the i18n WG on this change at https://github.com/w3c/html/issues/1039

**Pull request merged**